onnxruntime/core/providers/coreml/builders/impl/depthtospace_op_builder.cc

```diff
@@ -72,12 +72,6 @@ bool DepthToSpaceOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderI
     LOGS(logger, VERBOSE) << "DepthToSpace only supports 4d shape, input is " << input_size << "d shape.";
```

Comment:
is the check that input_size == 4 required here? the CoreML spec doesn't seem to require it. on the other hand, the ONNX DepthToSpace op seems to require it, so perhaps it is already checked.

Reply:
also, `input_rank` may be a clearer name

Reply:
Given this code is deciding if CoreML supports the node, I'd vote for removing the check or making it >= 4 if that is what CoreML supports. 

As a bonus if in the future ONNX DepthToSpace changes to allow 5D we don't need any code changes here.